src/sagas/deposit.js

```diff
@@ -191,47 +229,84 @@ export function* getBitcoinAddress() {
     /** @type Deposit */
     const deposit = yield select(state => state.deposit.deposit)
 
-    const btcAddress = yield deposit.bitcoinAddress
-
-    yield put({
-        type: DEPOSIT_BTC_ADDRESS,
-        payload: {
-            btcAddress,
-        }
-    })
+    try {
+        const btcAddress = yield deposit.bitcoinAddress
+        yield put({
+            type: DEPOSIT_BTC_ADDRESS,
+            payload: {
+                btcAddress,
+            }
+        })
+    } catch (error) {
+        yield put({
+            type: DEPOSIT_BTC_ADDRESS_ERROR,
+            payload: {
+                error: error.message,
+            }
+        })
+        return
+    }
 
-    const lotInSatoshis = yield call([deposit, deposit.getSatoshiLotSize])
-    const signerFeeTbtc = yield call([deposit, deposit.getSignerFeeTBTC])
-    const signerFeeInSatoshis = signerFeeTbtc.div(tbtc.satoshisPerTbtc)
-    yield put({
-        type: DEPOSIT_BTC_AMOUNTS,
-        payload: {
-            lotInSatoshis,
-            signerFeeInSatoshis,
-        }
-    })
+    try {
+        const lotInSatoshis = yield call([deposit, deposit.getSatoshiLotSize])
+        const signerFeeTbtc = yield call([deposit, deposit.getSignerFeeTBTC])
+        const signerFeeInSatoshis = signerFeeTbtc.div(tbtc.satoshisPerTbtc)
+        yield put({
+            type: DEPOSIT_BTC_AMOUNTS,
+            payload: {
+                lotInSatoshis,
+                signerFeeInSatoshis,
+            }
+        })
+    } catch (error) {
+        yield put({
+            type: DEPOSIT_BTC_AMOUNTS_ERROR,
+            payload: {
+                error: error.message,
+            }
+        })
+        return
+    }
 
     // goto
     yield put(navigateTo('/deposit/' + deposit.address + '/pay'))
+    yield* autoSubmitDepositProof()
 }
 
 export function* autoSubmitDepositProof() {
     /** @type Deposit */
     const deposit = yield select(state => state.deposit.deposit)
+    const didSubmitDepositProof =
+        yield select(state => state.deposit.didSubmitDepositProof)
```

Comment:
This is actually an interesting anchor for the FIXME work from above when we get to it.